block/local/adapter.go

```diff
@@ -172,6 +173,21 @@ func (l *Adapter) Get(obj block.ObjectPointer, _ int64) (reader io.ReadCloser, e
 	return f, nil
 }
 
+func (l *Adapter) Walk(walkOpt block.WalkOpts, walkFn block.WalkFunc) error {
+	p := filepath.Clean(path.Join(l.path, walkOpt.StorageNamespace, walkOpt.Prefix))
+	if err := filepath.Walk(p, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		return walkFn(p)
+	}); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
```suggestion
	return filepath.Walk(p, func(path string, info os.FileInfo, err error) error {
		return walkFn(p)
	})
```

Reply:
I still want to handle the inner error, the wrapped code is Done.